cipher/rsa/rsa2.go

```diff
@@ -0,0 +1,131 @@
+/*
+rsa2.go
+description: RSA encryption and decryption including key generation
+details: [RSA wiki](https://en.wikipedia.org/wiki/RSA_(cryptosystem))
+author(s): [ddaniel27](https://github.com/ddaniel27)
+*/
+package rsa
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"math/rand"
+
+	"github.com/TheAlgorithms/Go/math/gcd"
+	"github.com/TheAlgorithms/Go/math/lcm"
+	"github.com/TheAlgorithms/Go/math/modular"
+	"github.com/TheAlgorithms/Go/math/prime"
+)
+
+// rsa struct contains the public key, private key and modulus
+type rsa struct {
+	publicKey  uint64
+	privateKey uint64
+	modulus    uint64
+}
+
+// InitRSA initializes the RSA algorithm
+// returns the RSA object
+func InitRSA() *rsa {
+	rsa := new(rsa)
+	rsa.keyGen()
+	return rsa
+}
+
+// KeyGen generates a key for encryption and decryption with RSA
+// 1. Choose two large prime numbers (p and q) and compute n = p * q
+// 2. Compute the totient of n, lcm(p-1, q-1)
+// 3. Choose an integer e such that 1 < e < lcm(p-1, q-1) and gcd(e, lcm(p-1, q-1)) = 1
+// 4. Compute d such that d * e ≡ 1 (mod lcm(p-1, q-1))
+func (rsa *rsa) keyGen() {
```

Comment:
This functions seems to be only used in the `InitRSA` function, so you can inline it there.

Reply:
Done!